packages/@aws-cdk-containers/ecs-service-extensions/lib/extensions/subscribe.ts

```diff
@@ -0,0 +1,294 @@
+import * as ecs from '@aws-cdk/aws-ecs';
+import * as sns from '@aws-cdk/aws-sns';
+import * as subscription from '@aws-cdk/aws-sns-subscriptions';
+import * as sqs from '@aws-cdk/aws-sqs';
+import * as cdk from '@aws-cdk/core';
+import { Service } from '../service';
+import { Container } from './container';
+import { ContainerMutatingHook, ServiceExtension } from './extension-interfaces';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct } from '@aws-cdk/core';
+
+/**
+ * The settings for the subscribe extension.
+ */
+export interface SubscribeProps {
+  /**
+   * The list of topic subscriptions for this service.
+   *
+   * @default none
+   */
+  readonly topicSubscriptions?: TopicSubscriptionProps[];
+
+  /**
+   * The props for creating the default queue for this service.
+   * Please provide either the `eventsQueueProps` or the `eventsQueue`, but not both.
+   *
+   * @default none
+   */
+  readonly eventsQueueProps?: sqs.QueueProps;
+
+  /**
+   * The user-provided default queue for this service.
+   * Please provide either the `eventsQueueProps` or the `eventsQueue`, but not both.
+   *
+   * @default none
+   */
+  readonly eventsQueue?: sqs.IQueue;
+
+  /**
+   * The props for the default Dead Letter Queue.
+   *
+   * @default none
+   */
+  readonly eventsDeadLetterQueueProps?: DeadLetterQueueProps;
```

Comment:
I don't get why we need this field. Can't they just specify dead letter queue within `eventsQueueProps`?

Reply:
We decided to drop the queue and dead letter queue props completely as they were only being used to instantiate the SQS Queue and served no other purpose. So now, any queue that the user doesn't want to be created as default needs to be passed in as an input prop. This seems to largely simplify the experience as well.

Do let me know if that makes sense or if you have any suggestions!

Reply:
sgtm